tests/pipelines/controlnet/test_controlnet_sdxl.py

```diff
@@ -297,3 +297,20 @@ def test_stable_diffusion_xl_prompt_embeds(self):
 
         # make sure that it's equal
         assert np.abs(image_slice_1.flatten() - image_slice_2.flatten()).max() < 1e-4
+
+
+class ControlNetPipelineSDXLGuessModeFastTests(ControlNetPipelineSDXLFastTests):
```

Comment:
Can we test all four scenarios here? 
* guess_mode == True & CFG == True
* guess_model ==False and CFG == True
* guess_model == False & CFG == False
* guess_mode == True and CFG == False

I found the logic a little bit complex so let's first add test here to make sure it works as expected here first

Reply:
Hi, If you have any idea on how we could simplify the logic, it would be welcome!

Regarding tests, so you think it is sufficient to run functional tests (i.e. run pipeline with all combinations of parameters and make sure it does not fail) or we need to run heavy tests with real checkpoint loading and checking the outputs?

Reply:
Functional tests are fine here.

Reply:
@sayakpaul
Added tests for all combinations of params: https://github.com/huggingface/diffusers/pull/4155/commits/87b7cfe780e27e36d06241967e63bf1c72faf15f

I was not sure how to add multiple options given `unittest` limited functionality in parametrisation, so feel free to suggest better options

Reply:
I think we can make use of `parameterized` here no? A couple of our test cases use it. 

Let me know.

Reply:
> Let me know.

I have already used it, please have a look at the commit https://github.com/huggingface/diffusers/commit/87b7cfe780e27e36d06241967e63bf1c72faf15f.

Though `parameterized` had a bug related to classes inheritance, I have used a workaround, but I don't like it too much

Reply:
I would just make 4 tests - and each with its own expected slices